models/appModel.js

```diff
@@ -0,0 +1,40 @@
+var AppModel = Backbone.Model.extend({
+  constructor: function() {
+    this.videos = new VideoCollection();
+    this.currentVideo = new VideoModel();
+  },
+
+  getVideos: function(query) {
+    var self = this;
+    Backbone.ajax({
+      method: "GET",
+      url: "https://www.googleapis.com/youtube/v3/search/?part=snippet&q=" + query + "&type=video&key=AIzaSyC5_RVmMP3hnqrdStrJ7vLFdAPxjNB_2sY",
+      dataType: "json",
+      success: function(data) {
+        //If the videos collection is empty, it'll add models to it, otherwise, it'll modify existing models.
+        if (self.videos.length == 0) {
+          for (var i = 0; i < data.items.length; i++) {
+            var newModel = new VideoModel({
+              videoId: data.items[i].id.videoId,
+              title: data.items[i].snippet.title,
+              description: data.items[i].snippet.description,
+              thumbnailURL: data.items[i].snippet.thumbnails.default.url
+            });
+            self.videos.add(newModel);
+          }
+        }
+        else {
```

Comment:
So in practice, you usually treat the data coming from the server as authoritative and you'd never need to do this.  I understand that it seems like you're being more efficient, but really you should just always do the if condition in this if/else.